apm-agent-api/src/main/java/co/elastic/apm/api/Span.java

```diff
@@ -466,4 +466,15 @@ public interface Span {
      */
     @Nonnull
     Span setDestinationService(@Nullable String resource);
```

Comment:
Should this method be deprecated? What's our recommendation for users for when to use `setDestinationService` vs `setServiceTarget`?

Reply:
Yes, we can deprecate it, the behavior will be:
- `setDestinationResource(null)` behavior is preserved: resource and service target fields are dropped
- `setDestinationResource("value")` will set `resource` = `"value"`, `service.target.type` = `""` (empty value), `service.target.name` = `"value"` : just reusing the behavior defined on server [here](https://github.com/elastic/apm/blob/c19f4e9de1bb46961169b63e5aa2d52feb4d479d/specs/agents/tracing-spans-service-target.md#phase-3)
- replacing `setDestinationResource("value")` with `setServiceTarget("type", "value")` will force the user to pick a value for `type` and will change the resource to `"type/value"`, if they really want to keep it as previously they can combine it with a call to `setDestinationResource` to preserve compatibility.

Reply:
I've removed the doc + deprecated the API, the strategy with empty `""` type have also been added to the specification.